boot/bootutil/include/bootutil/zephyr/mcuboot_shared.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2023, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <stddef.h>
+#include <bootutil/image.h>
+
+enum mcuboot_configuration_version {
+    MCUBOOT_CONFIGURATION_VERSION_1 = 1,
+};
+
+enum mcuboot_mode {
+    MCUBOOT_MODE_SINGLE_SLOT,
+    MCUBOOT_MODE_SWAP_USING_SCRATCH,
+    MCUBOOT_MODE_UPGRADE_ONLY,
+    MCUBOOT_MODE_SWAP_USING_MOVE,
+    MCUBOOT_MODE_DIRECT_XIP,
+    MCUBOOT_MODE_RAM_LOAD
+};
+
+enum mcuboot_signature_type {
+    MCUBOOT_SIGNATURE_TYPE_NONE,
+    MCUBOOT_SIGNATURE_TYPE_RSA,
+    MCUBOOT_SIGNATURE_TYPE_ECDSA_P256,
+    MCUBOOT_SIGNATURE_TYPE_ED25519
+};
+
+enum mcuboot_recovery_mode {
+    MCUBOOT_RECOVERY_MODE_NONE,
+    MCUBOOT_RECOVERY_MODE_SERIAL_RECOVERY,
+    MCUBOOT_RECOVERY_MODE_DFU,
+};
+
+struct mcuboot_configuration {
+    /** Version of this configuration struct #mcuboot_configuration_version */
+    uint8_t configuration_version;
```

Comment:
Don't you want to have version + structure size header first?
This way the other side can read the header first and then decide whether it can process it, by the version, and how much data to read and how to cast it, by the size.

Reply:
Wouldn't be able to use it like this, unless you mean the put the version and size at the end of the struct? Because the retention area size is fixed, the dts would need to be updated, so if the bootloader used version 1 and application used version 2 which were both different sizes, there would need to be a way to determine something like that from a common area

Reply:
Read through the docs and updated as per requirements